modules/lti-service-api/src/main/java/org/opencastproject/lti/service/api/LtiService.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ *
+ * The Apereo Foundation licenses this file to you under the Educational
+ * Community License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+package org.opencastproject.lti.service.api;
+
+import org.opencastproject.security.api.UnauthorizedException;
+import org.opencastproject.util.NotFoundException;
+
+import java.util.List;
+
+/**
+ * Interface for implementing functionality available in the LTI GUI
+ */
+public interface LtiService {
+  String JOB_TYPE = "org.opencastproject.lti.service";
+
+  /**
+   * List currently running jobs for the series
+   * @param seriesId ID of the series
+   * @return A list of jobs
+   */
+  List<LtiJob> listJobs(String seriesId);
+
+  /**
+   * Upload a new event or update existing event's metadata
+   * @param file File to upload
+   * @param captions Subtitles file
+   * @param eventId ID of the event (can be <code>null</code> for new events)
+   * @param seriesId ID of the series
+   * @param metadataJson Metadata for the event as JSON string
+   */
+  void upsertEvent(
```

Comment:
Typo? `uploadEvent`?

Reply:
Not really, the function inserts or updates, thus upserts. It's got a Wiktionary entry, even: https://en.wiktionary.org/wiki/upsert :)